### ISSUE TITLE
Segfault during startup after upgrading to 23.12.1.1368

### ISSUE THREAD

**Issue by campi01** [ClickHouse/ClickHouse#58451]
Hello,
we are using CH solely as a backend for Graphite Clickhouse [https://github.com/go-graphite/graphite-clickhouse](url), in a 14 node cluster (7 shards with 2 replicas), running on Debian 10. After upgrading from version 23.11.2.11 to version 23.12.1.1368, we get a segfault during startup of clickhouse-server. Upgrading to version 23.11.3.23 works fine.

**How to reproduce**
Upgrade clickhouse-server to latest version 23.12.1.1368, and restart the service.

* Which ClickHouse server version to use
23.12.1.1368

*Stack trace

```
2024.01.03 10:55:16.257352 [ 5207 ] {} <Fatal> BaseDaemon: ########## Short fault info ############
2024.01.03 10:55:16.257421 [ 5207 ] {} <Fatal> BaseDaemon: (version 23.12.1.1368 (official build), build id: 19BC8CB87C441A86F879D18F5C18F17BA08526EE, git hash: a2faa65b080a587026c86844f3a20c74d23a86f8) (from thread 4678) Received signal 11
2024.01.03 10:55:16.257471 [ 5207 ] {} <Fatal> BaseDaemon: Signal description: Segmentation fault
2024.01.03 10:55:16.257498 [ 5207 ] {} <Fatal> BaseDaemon: Address: NULL pointer. Access: read. Address not mapped to object.
2024.01.03 10:55:16.257520 [ 5207 ] {} <Fatal> BaseDaemon: Stack trace: 0x0000000012429abc 0x00000000124230bb 0x0000000011c1132a 0x000000001237eae9 0x000000001034e13f 0x000000001034f191 0x000000000c7be158 0x00007f6841afffa3 0x00007f6841a3106f
2024.01.03 10:55:16.257577 [ 5207 ] {} <Fatal> BaseDaemon: ########################################
2024.01.03 10:55:16.257609 [ 5207 ] {} <Fatal> BaseDaemon: (version 23.12.1.1368 (official build), build id: 19BC8CB87C441A86F879D18F5C18F17BA08526EE, git hash: a2faa65b080a587026c86844f3a20c74d23a86f8) (from thread 4678) (no query) Received signal Segmentation fault (11)
2024.01.03 10:55:16.257632 [ 5207 ] {} <Fatal> BaseDaemon: Address: NULL pointer. Access: read. Address not mapped to object.
2024.01.03 10:55:16.257647 [ 5207 ] {} <Fatal> BaseDaemon: Stack trace: 0x0000000012429abc 0x00000000124230bb 0x0000000011c1132a 0x000000001237eae9 0x000000001034e13f 0x000000001034f191 0x000000000c7be158 0x00007f6841afffa3 0x00007f6841a3106f
2024.01.03 10:55:16.257763 [ 5207 ] {} <Fatal> BaseDaemon: 2. DB::Graphite::Pattern::updateHash(SipHash&) const @ 0x0000000012429abc in /usr/bin/clickhouse
2024.01.03 10:55:16.257815 [ 5207 ] {} <Fatal> BaseDaemon: 3. DB::ReplicatedMergeTreeTableMetadata::ReplicatedMergeTreeTableMetadata(DB::MergeTreeData const&, std::shared_ptr<DB::StorageInMemoryMetadata const> const&) @ 0x00000000124230bb in /usr/bin/clickhouse
2024.01.03 10:55:16.259803 [ 5207 ] {} <Fatal> BaseDaemon: 4. DB::StorageReplicatedMergeTree::checkTableStructure(String const&, std::shared_ptr<DB::StorageInMemoryMetadata const> const&, bool) @ 0x0000000011c1132a in /usr/bin/clickhouse
2024.01.03 10:55:16.259827 [ 5207 ] {} <Fatal> BaseDaemon: 5. DB::ReplicatedMergeTreeAttachThread::run() @ 0x000000001237eae9 in /usr/bin/clickhouse
2024.01.03 10:55:16.259852 [ 5207 ] {} <Fatal> BaseDaemon: 6. DB::BackgroundSchedulePool::threadFunction() @ 0x000000001034e13f in /usr/bin/clickhouse
2024.01.03 10:55:16.259893 [ 5207 ] {} <Fatal> BaseDaemon: 7. void std::__function::__policy_invoker<void ()>::__call_impl<std::__function::__default_alloc_func<ThreadFromGlobalPoolImpl<false>::ThreadFromGlobalPoolImpl<DB::BackgroundSchedulePool::BackgroundSchedulePool(unsigned long, StrongTypedef<unsigned long, CurrentMetrics::MetricTag>, StrongTypedef<unsigned long, CurrentMetrics::MetricTag>, char const*)::$_0>(DB::BackgroundSchedulePool::BackgroundSchedulePool(unsigned long, StrongTypedef<unsigned long, CurrentMetrics::MetricTag>, StrongTypedef<unsigned long, CurrentMetrics::MetricTag>, char const*)::$_0&&)::'lambda'(), void ()>>(std::__function::__policy_storage const*) @ 0x000000001034f191 in /usr/bin/clickhouse
2024.01.03 10:55:16.259931 [ 5207 ] {} <Fatal> BaseDaemon: 8. void* std::__thread_proxy[abi:v15000]<std::tuple<std::unique_ptr<std::__thread_struct, std::default_delete<std::__thread_struct>>, void ThreadPoolImpl<std::thread>::scheduleImpl<void>(std::function<void ()>, Priority, std::optional<unsigned long>, bool)::'lambda0'()>>(void*) @ 0x000000000c7be158 in /usr/bin/clickhouse
2024.01.03 10:55:16.260665 [ 5207 ] {} <Fatal> BaseDaemon: 9. start_thread @ 0x0000000000007fa3 in /lib/x86_64-linux-gnu/libpthread-2.28.so
2024.01.03 10:55:16.260687 [ 5207 ] {} <Fatal> BaseDaemon: 10. ? @ 0x00000000000f906f in /lib/x86_64-linux-gnu/libc-2.28.so
2024.01.03 10:55:16.597142 [ 5207 ] {} <Fatal> BaseDaemon: Integrity check of the executable successfully passed (checksum: E62A564F8077C0AF79E8E5726EE71488)
2024.01.03 10:55:16.600962 [ 5207 ] {} <Fatal> BaseDaemon: Report this error to https://github.com/ClickHouse/ClickHouse/issues
2024.01.03 10:55:20.708171 [ 4444 ] {} <Fatal> Application: Child process was terminated by signal 11.
```

I have also tried searching through the logs for thread 4678, but there is nothing in the logs. Please let me know if you require any further information.

**Comment by canhld94** [comment]
Relates to my change in #56833, will fix asap

**Comment by canhld94** [comment]
@campi01 just to be more clear, your graphite configuration contains pattern with no aggregate function, right?

**Comment by campi01** [comment]
> @campi01 just to be more clear, your graphite configuration contains pattern with no aggregate function, right?

That is correct, though we have a function configured in the default section. Here is the full config:

```
<yandex>
  <graphite_rollup>
    <pattern>
      <regexp>\.min$</regexp>
      <function>min</function>
    </pattern>
    <pattern>
      <regexp>\.max$</regexp>
      <function>max</function>
    </pattern>
    <pattern>
      <regexp>\.count$</regexp>
      <function>sum</function>
    </pattern>
    <pattern>
      <regexp>^(example|strings)\.[^.]+</regexp>
      <retention>
        <age>0</age>
        <precision>60</precision>
      </retention>
      <retention>
        <age>252000</age>
        <precision>86400</precision>
      </retention>
    </pattern>
    <default>
      <function>avg</function>
      <retention>
        <age>0</age>
        <precision>60</precision>
      </retention>
      <retention>
        <age>3600</age>
        <precision>86400</precision>
      </retention>
    </default>
  </graphite_rollup>
</yandex>
```

**Comment by canhld94** [comment]
> > @campi01 just to be more clear, your graphite configuration contains pattern with no aggregate function, right?
> 
> 
> 
> That is correct, though we have a function configured in the default section. Here is the full config:
> 
> 
> 
> ```
> 
> <yandex>
> 
>   <graphite_rollup>
> 
>     <pattern>
> 
>       <regexp>\.min$</regexp>
> 
>       <function>min</function>
> 
>     </pattern>
> 
>     <pattern>
> 
>       <regexp>\.max$</regexp>
> 
>       <function>max</function>
> 
>     </pattern>
> 
>     <pattern>
> 
>       <regexp>\.count$</regexp>
> 
>       <function>sum</function>
> 
>     </pattern>
> 
>     <pattern>
> 
>       <regexp>^(example|strings)\.[^.]+</regexp>
> 
>       <retention>
> 
>         <age>0</age>
> 
>         <precision>60</precision>
> 
>       </retention>
> 
>       <retention>
> 
>         <age>252000</age>
> 
>         <precision>86400</precision>
> 
>       </retention>
> 
>     </pattern>
> 
>     <default>
> 
>       <function>avg</function>
> 
>       <retention>
> 
>         <age>0</age>
> 
>         <precision>60</precision>
> 
>       </retention>
> 
>       <retention>
> 
>         <age>3600</age>
> 
>         <precision>86400</precision>
> 
>       </retention>
> 
>     </default>
> 
>   </graphite_rollup>
> 
> </yandex>
> 
> ```

Thanks, it should be fixed in #58453
